compiler/native/expand.go

```diff
@@ -47,11 +47,18 @@ func (c *client) ExpandStages(s *yaml.Build, tmpls map[string]*yaml.Template, r
 
 // ExpandSteps injects the template for each
 // templated step in a yaml configuration.
-func (c *client) ExpandSteps(s *yaml.Build, tmpls map[string]*yaml.Template, r *pipeline.RuleData) (*yaml.Build, error) {
+func (c *client) ExpandSteps(s *yaml.Build, tmpls map[string]*yaml.Template, r *pipeline.RuleData, depth int) (*yaml.Build, error) {
 	if len(tmpls) == 0 {
 		return s, nil
 	}
 
+	// if max template depth has been reached, return to avoid circular dependencies
```

Comment:
```suggestion
	// return if max template depth has been reached
```
minor, but it makes it sound like it's exclusively about circular dependencies.